onnx_xla/translations/translate_reshape.cc

```diff
@@ -0,0 +1,83 @@
+#include "onnx_xla/operator_registry.h"
+
+namespace onnx_xla {
+// TODO: ENFORCE_EQ macro
+// Takes in data tensor and shape input
+// Error if shape input is dynamic
+// Parses ONNX shape input (can have 0's and one -1)
+// Produces new_sizes attribute to make XlaOp for Reshape
+onnxStatus translateReshape(const Node& n,
+                            XlaBuilder& builder,
+                            ValueOpMap& valueToOp,
+                            const ValueLiteralMap& valueToLiteral) {
+  // Check if shape input is static
+  auto shapeDataIt = valueToLiteral.find(n.inputs().at(1));
+  if (shapeDataIt == valueToLiteral.end()) {  // TODO: ENFORCE_EQ
+    std::cerr << "Reshape operatore having dynamic shape input is not supported"
+              << std::endl;
+    return ONNXIFI_STATUS_UNSUPPORTED_VERSION;
+  }
+  // Compute # of elements in input data tensor (check validity of shapes)
+  const std::vector<Dimension>& originalShape = n.inputs().at(0)->sizes();
+  int64_t numElements = 1;
+  for (const auto& dimension : originalShape) {
+    if (!dimension.is_int) {
+      std::cerr << "Reshape operator input data shape is not known"
+                << std::endl;
+      return ONNXIFI_STATUS_UNSUPPORTED_SHAPE;
+    }
+    numElements *= dimension.dim;
+  }
+
+  // Use static ONNX target shape data (from the constant literal constructed)
+  // to build XLA target shape data and compute xlaProduct (product of shape
+  // constructed)
+  const tensorflow::gtl::ArraySlice<int64> onnxOperatorTargetShape =
+      shapeDataIt->second->data<int64>();
+  std::vector<int64> xlaOperatorTargetShape;
+  int64 xlaProduct = 1;
```

Comment:
why not int64_t for this one?

Reply:
Taking product over vector of int64, so thought int64 more appropriate than int64_t

Reply:
int64 is not standard c++ type I think. Is it possible to change to use int64_t everywhere (or is it tensorflow forcing us to use int64)?

Reply:
Tensorflow is complaining if I call on their functions with int64_t instead of int64. Some places where this can cause problems: anywhere where they convert between their primitive and native types, using their ArraySlice container (unable to cause between vector<int64_t> and their ArraySlice<int64>). I think the best approach is to work with C++ standard but use their native types whenever I have to call their functions. In this case xlaProduct is for my internal use so I'll change that to int64_t